sdk/servicebus/service-bus/src/util/atomXmlHelper.ts

```diff
@@ -0,0 +1,382 @@
+      const firstIndexOfRulesDelimiter = pathname.indexOf("/Rules/");
+      entry[nameProperties[0]] = pathname.substring(
+        firstIndexOfDelimiter + 1,
+        lastIndexOfSubscriptionsDelimiter
+      );
+      entry[nameProperties[1]] = pathname.substring(
+        lastIndexOfSubscriptionsDelimiter + 15,
+        firstIndexOfRulesDelimiter
+      );
+      entry[nameProperties[2]] = pathname.substring(firstIndexOfRulesDelimiter + 7);
+    } else if (pathname.match("(.*)/(.*)/Subscriptions/(.*)")) {
+      const lastIndexOfSubscriptionsDelimiter = pathname.lastIndexOf("/Subscriptions/");
+      entry[nameProperties[0]] = pathname.substring(
+        firstIndexOfDelimiter + 1,
+        lastIndexOfSubscriptionsDelimiter
+      );
+      entry[nameProperties[1]] = pathname.substring(lastIndexOfSubscriptionsDelimiter + 15);
+    } else if (pathname.match("(.*)/(.*)")) {
+      entry[nameProperties[0]] = pathname.substring(firstIndexOfDelimiter + 1);
+    }
+  }
+}
+
+/**
+ *
+ * Utility to help construct the normalized `RestError` object based on given `errorBody`
+ * data and other data present in the received `response` object.
+ *
+ * @param errorBody
+ * @param response
+ */
+export function buildError(errorBody: any, response: HttpOperationResponse): RestError {
+  const normalizedError: any = {};
+  const odataErrorFormat = !!errorBody["odata.error"];
+  const errorProperties =
+    errorBody.Error || errorBody.error || errorBody["odata.error"] || errorBody;
+  let errorMessage;
+
+  if (typeof errorBody === "string") {
+    errorMessage = errorBody;
+  } else {
+    if (odataErrorFormat) {
+      Object.keys(errorProperties).forEach((property: string) => {
+        let value = null;
+        if (
+          property === Constants.ODATA_ERROR_MESSAGE &&
+          typeof errorProperties[Constants.ODATA_ERROR_MESSAGE] !== "string"
+        ) {
+          if (errorProperties[Constants.ODATA_ERROR_MESSAGE][Constants.ODATA_ERROR_MESSAGE_VALUE]) {
+            value =
+              errorProperties[Constants.ODATA_ERROR_MESSAGE][Constants.ODATA_ERROR_MESSAGE_VALUE];
+          } else {
+            value = "missing value in the message property of the odata error format";
+          }
+        } else {
+          value = errorProperties[property];
+        }
+        normalizedError[property.toLowerCase()] = value;
+      });
+    } else {
+      Object.keys(errorProperties).forEach((property: any) => {
+        {
+          let value = null;
+          if (property !== Constants.XML_METADATA_MARKER) {
+            if (
+              errorProperties[property] &&
+              errorProperties[property][Constants.XML_VALUE_MARKER]
+            ) {
+              value = errorProperties[property][Constants.XML_VALUE_MARKER];
+            } else {
+              value = errorProperties[property];
+            }
+            normalizedError[property.toLowerCase()] = value;
+          }
+        }
+      });
+    }
```

Comment:
I believe I mentioned this when this code was part of the core-http PR and it might have gotten lost when we moved the code to service bus, so I'll repeat here.

There is a scope of improving the code here for readability and maintainence purposes.

```javascript
Object.keys(errorProperties).forEach((property: any) => {
          let value = errorProperties[property];
          if (property !== Constants.XML_METADATA_MARKER) {
            if (value && value[Constants.XML_VALUE_MARKER]) {
              value = value[Constants.XML_VALUE_MARKER];
            }
            normalizedError[property.toLowerCase()] = value;
          }
      });
```

Reply:
Updated